### PR TITLE
Basic auto profiler

### PR DESCRIPTION
**Pull request by AnturK**
Very simple stuff, just dumps whole round profile at shutdown.

### CODE REVIEW

**Comment by Jalleo** [comment]
This is going to help longterm with reference to the statbus to get context of what might be performing badly.

File: config/config.txt [hunk 1]
```diff
@@ -502,3 +502,7 @@ DEFAULT_VIEW 15x15
 ## The alternative square viewport size if you're using a widescreen view size
 ## You probably shouldn't ever be changing this, but it's here if you want to.
 DEFAULT_VIEW_SQUARE 15x15
+
+
+## Enable automatic profiling
```
Comment (MrPerson) [inline config/config.txt]: You might want to add a comment that this is only supported in 1506 or above

File: code/controllers/subsystem/profiler.dm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/datum/controller/subsystem/profiler/Shutdown()
+	if(CONFIG_GET(flag/auto_profile))
+		DumpFile()
+	return ..()
+
+/datum/controller/subsystem/profiler/proc/StartProfiling()
+#if DM_BUILD < 1506
+	stack_trace("Auto profiling unsupported on this byond version")
+#else
```
Comment (ninjanomnom) [inline code/controllers/subsystem/profiler.dm]: Disable the config flag again after

File: code/controllers/subsystem/profiler.dm [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	world.Profile(PROFILE_START)
```
Comment (ninjanomnom) [inline code/controllers/subsystem/profiler.dm]: It should be starting as early as possible if we have automatic profiling enabled, idealy as the first thing in /world/New so we can get profiling data from startup. I don't know if a regular config flag is good for this

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/profiler.dm (hunk 3 above)
Comment (AnturK) [reply code/controllers/subsystem/profiler.dm]: I guess i can start the profiler early if possible and just stop it during init if the flag is unset, not very clean solution though.

File: code/controllers/subsystem/profiler.dm [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+/datum/controller/subsystem/profiler/proc/StartProfiling()
+#if DM_BUILD < 1506 || DM_VERSION < 513
+	stack_trace("Auto profiling unsupported on this byond version")
+	CONFIG_SET(flag/auto_profile, FALSE)
+#else
+	world.Profile(PROFILE_START)
+#endif
+
+/datum/controller/subsystem/profiler/proc/StopProfiling()
+#if DM_BUILD > 1506 && DM_VERSION > 513
```
Comment (ninjanomnom) [inline code/controllers/subsystem/profiler.dm]:
```suggestion
#if DM_BUILD >= 1506 && DM_VERSION >= 513
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/profiler.dm (hunk 4 above)
Comment (AnturK) [reply code/controllers/subsystem/profiler.dm]: whoops

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**